src/main/java/online/shop/impl/ProductDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package online.shop.impl;
```

Comment:
Maybe it would be better to have DaoImpl classes in a package "impl" in "dao", so that DAO implementations are separated from service implementations?